usage/rest-client/src/main/java/org/apache/brooklyn/rest/client/BrooklynApi.java

```diff
@@ -114,11 +150,23 @@ public BrooklynApi(String endpoint, @Nullable Credentials credentials) {
         }
     }
 
+    /**
+     * @deprecated since 0.9.0. Use {@link BrooklynApiFactory#getBrooklynApi(java.net.URL, String, String)} instead
+     * @see org.apache.brooklyn.rest.client.DefaultBrooklynApiFactory
+     * @see org.apache.brooklyn.rest.client.PoolingBrooklynApiFactory
+     */
+    @Deprecated
```

Comment:
Good point. This one is deprecated but the next one is not. The idea was that if you want change implementation then probably you just have to write another factory (and use the next constructor) :)